backend/TraderX/src/main/java/cmpe451/group6/rest/transaction/controller/TransactionController.java

```diff
@@ -0,0 +1,110 @@
+package cmpe451.group6.rest.transaction.controller;
+
+import cmpe451.group6.Util;
+import cmpe451.group6.rest.transaction.dto.TransactionDTO;
+import cmpe451.group6.rest.transaction.model.Transaction;
+import cmpe451.group6.rest.transaction.service.TransactionService;
+import io.swagger.annotations.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Date;
+import java.util.List;
+
+import static cmpe451.group6.authorization.exception.GlobalExceptionHandlerController.GENERIC_ERROR_RESPONSE;
+
+@RestController
+@RequestMapping("/transaction")
+@Api(tags = "Transaction")
+public class TransactionController {
+
+    @Autowired
+    private TransactionService transactionService;
+
+    @Autowired
+    private Util util;
+
+    @GetMapping(value = "/{username}")
+    @ResponseStatus(HttpStatus.OK)
+    @PreAuthorize("hasRole('ROLE_ADMIN') or hasRole('ROLE_BASIC') or hasRole('ROLE_TRADER')")
+    @ApiOperation(value = "Gets transactions of an user", response = Transaction.class)
+    @ApiResponses(value = { @ApiResponse(code = 417, message = "No such an equipment found.") })
+    public List<Transaction> getTransactionsByUser(@ApiParam("User Name") @PathVariable String username,
+            HttpServletRequest req) {
+        return transactionService.getTransactionsByUser(username, util.unwrapUsername(req));
+    }
+
+    @GetMapping(value = "/{code}")
+    @ResponseStatus(HttpStatus.OK)
+    @ApiOperation(value = "Returns transactions on specified equipment code (no auth required)", response = TransactionDTO.class)
+    @ApiResponses(value = { @ApiResponse(code = 400, message = GENERIC_ERROR_RESPONSE) })
+    public List<TransactionDTO> getTransactionsByCode(@ApiParam("User Name") @PathVariable String code) {
+        return transactionService.getTransactionsByCode(code);
+    }
+
+    @GetMapping(value = "/all")
+    @ResponseStatus(HttpStatus.OK)
+    @ApiOperation(value = "Returns all transactions (no auth required)", response = TransactionDTO.class)
+    @ApiResponses(value = { @ApiResponse(code = 400, message = GENERIC_ERROR_RESPONSE) })
+    public List<TransactionDTO> getAllTransactions() {
+        return transactionService.getTransactions();
+    }
+
+    @GetMapping(value = "/byDate")
+    @ResponseStatus(HttpStatus.OK)
+    @ApiOperation(value = "Returns transactions in between start date and end date (no auth required)", response = TransactionDTO.class)
+    @ApiResponses(value = { @ApiResponse(code = 400, message = GENERIC_ERROR_RESPONSE) })
+    public List<TransactionDTO> getTransactionsByDateBetween(@ApiParam("Start Date") @RequestParam Date start,
+            @ApiParam("End Date") @RequestParam Date end) {
```

Comment:
In which format those dates have to be sent? We better clarify as timestamps.

Reply:
The format is mentioned now.